src/drivers/intel/baytrail/ssp.c

```diff
@@ -165,7 +165,7 @@ static int ssp_set_config(struct dai *dai,
 	sspsp = SSPSP_ETDS; /* last value (bit 0) */
 
 	ssp->config = *config;
-	ssp->params = config->ssp;
+	ssp->params = config->dai_data[0].ssp;
```

Comment:
@plbossart should we change this all to config->dai_data[config->config_index] instead of 0 to make it ready when the config_index would be something other than 0?

Reply:
i wanted to do this in follow-up patches when we have support for this at the kernel level. For now it's simpler.
And I am not sure that we want to have multiple configurations for all DAI types anyways, so it's rather a case-by-case change, not a blanket one.

Reply:
look good in that case @plbossart . we've still  got some build failures on a couple of platforms